### PR TITLE
- [vulkan-loader] fix clang-cl

### PR DESCRIPTION
**Pull request by SSE4**
closes: #9964

Specify library name and version:  **vulkan-loader/all**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/vulkan-loader/all/conanfile.py [hunk 1]
```diff
@@ -42,6 +42,10 @@ def _source_subfolder(self):
     def _is_mingw(self):
         return self.settings.os == "Windows" and self.settings.compiler == "gcc"
 
+    @property
+    def _is_clang_cl(self):
+        return self.settings.os == "Windows" and self.settings.compiler == "clang"
+
```
Comment (SpaceIm) [inline recipes/vulkan-loader/all/conanfile.py:48]:
```suggestion
```

File: recipes/vulkan-loader/all/conanfile.py [hunk 2]
```diff
@@ -100,6 +104,8 @@ def _patch_sources(self):
         tools.replace_in_file(os.path.join(self._source_subfolder, "loader", "CMakeLists.txt"),
                               "if(${configuration} MATCHES \"/MD\")",
                               "if(FALSE)")
+        if self._is_clang_cl:
+            tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"), '/WX', '')
```
Comment (SpaceIm) [inline recipes/vulkan-loader/all/conanfile.py:108]:
```suggestion
        # No warnings as errors
        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"), '/WX', '')
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]
You can remove /WX unconditionally.

**Comment by SSE4** [comment]
> You can remove /WX unconditionally.

yes, it shouldn't break anyone

**Comment by SpaceIm** [comment]
> > You can remove /WX unconditionally.
> 
> yes, it shouldn't break anyone

I would even say that it could avoid build errors of old vulkan-loader versions with recent msvc. `/WX` & `-Werror` should always be disabled/removed in conan-center recipes.

**Review by AndreyMlashkin — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Review by AndreyMlashkin — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

**Pull request merged**